javascript/ql/src/Security/CWE-020/IncompleteHostnameRegExp.ql

```diff
@@ -13,16 +13,41 @@
 import javascript
 
 /**
- * A taint tracking configuration for incomplete hostname regular expressions sources.
+ * Gets a node whose value may flow (inter-procedurally) to a position where it is interpreted
+ * as a regular expression.
  */
-class Configuration extends TaintTracking::Configuration {
-  Configuration() { this = "IncompleteHostnameRegExpTracking" }
+DataFlow::Node regExpSource(DataFlow::Node re, DataFlow::TypeBackTracker t) {
+  t.start() and
+  re = result and
+  isInterpretedAsRegExp(result)
+  or
+  exists(DataFlow::TypeBackTracker t2, DataFlow::Node succ | succ = regExpSource(re, t2) |
+    exists(DataFlow::StepSummary summary |
+      DataFlow::StepSummary::smallstep(result, succ, summary) and
+      t = t2.prepend(summary)
+      or
+      result = succ.getAPredecessor() and
+      t = t2
+      or
+      any(TaintTracking::AdditionalTaintStep dts).step(result, succ) and
+      (t.continue() = t2 or t = t2)
```

Comment:
`t.continue()` returns either `t` or nothing, so it seems one of these disjuncts is unnecessary. AFAICT this is the same as `t = t2`.

Reply:
Yes, that is a mistake.